test/test_model.py

```diff
@@ -3596,5 +3596,27 @@ def test_set_ports_noop(self):
         ])
 
 
+class TestUnit(unittest.TestCase):
```

Comment:
Yeah, there's a *lot* stuffed into `TestModel` right now. I think (starting to) separate them out is the right call -- thanks.